networkx/generators/directed.py

```diff
@@ -241,13 +241,14 @@ def scale_free_graph(n, alpha=0.41, beta=0.54, gamma=0.05, delta_in=0.2,
            Discrete Algorithms, 132--139, 2003.
     """
 
+    # TODO
     def _choose_node(G,distribution,delta):
         cumsum=0.0
         # normalization 
-        psum=float(sum(distribution.values()))+float(delta)*len(distribution)
-        r=random.random()
-        for i in range(0,len(distribution)):
-            cumsum+=(distribution[i]+delta)/psum
+        psum = float(sum(distribution.values())) + float(delta) * len(G)
+        r = random.random()
+        for i in range(0, len(G)):
+            cumsum += (distribution[i] + delta) / psum
```

Comment:
@dschult This works :) 
We don't need to move the calculation outside for `psum`. But should we? We can make it `psum_in` for `delta_in` and `psum_out` for `delta_out`. This will prevent calculating `psum` every time `_choose_node` is called. We can pass `psum_in` and `psum_out` as arguments to it.